backend/api/views/dataset.py

```diff
@@ -26,30 +30,73 @@ def list(self, request):
         queryset = Dataset.objects.annotate(Count("files")).select_related(
             "dataset_type"
         )
+
         serializer = self.serializer_class(queryset, many=True)
-        return Response(serializer.data)
+        stack_message = messages.get_messages(request)
+        flash_messages = []
+        for message in stack_message:
+            flash_messages.append({"message": message.message, "tags": message.tags})
+        resultat = {
+            "flash_messages": flash_messages,
+            "result": serializer.data,
+        }
+        return Response(resultat)
 
     @action(detail=False)
+    def list_to_import(self, request):
+        """list dataset in datasets.csv"""
+        dataset_names = Dataset.objects.values_list("name", flat=True)
+        csv_dataset_names = []
+        new_datasets = []
+
+        # Check for new datasets
+        try:
+            with open(
+                settings.DATASET_IMPORT_FOLDER / "datasets.csv", encoding="utf-8"
+            ) as csvfile:
+                for dataset in csv.DictReader(csvfile):
+                    csv_dataset_names.append(dataset["name"])
+                    if dataset["name"] not in dataset_names:
+                        new_datasets.append(dataset)
+        except FileNotFoundError as error:
+            capture_exception(error)
+            messages.add_message(request, messages.ERROR, f"File Not Found : {error}")
+            return redirect(reverse("dataset-list"))
+
+        return Response(new_datasets)
+
+    @action(detail=False, methods=["POST"])
     def datawork_import(self, request):
         """Import new datasets from datawork"""
         if not request.user.is_staff:
             return HttpResponse("Unauthorized", status=403)
 
         try:
-            new_datasets = datawork_import(importer=request.user)
+            new_datasets = datawork_import(
+                dataset_checked=request.data["dataset_checked"],
+                importer=request.user,
+            )
         except FileNotFoundError as error:
             capture_exception(error)
-            return HttpResponse(error, status=400)
+            messages.add_message(request, messages.ERROR, f"File Not Found : {error}")
+            return redirect(reverse("dataset-list"))
         except PermissionError as error:
             capture_exception(error)
-            return HttpResponse(error, status=400)
```

Comment:
Looking at the tests I'm guessing your new system returns a 200 response ? Why though ? For a REST API a 400 response is much more semantically appropriate and you should be able to manage it in the front same as a 200 :thinking: